frontend/vite.config.ts

```diff
@@ -30,5 +30,27 @@ export default defineConfig(() => {
         '@public': path.resolve(__dirname, './public'),
       },
     },
+    test: {
+      globals: true,
+      environment: 'happy-dom',
```

Comment:
What is the reason behind happy-dom instead of jsdom?
I never used happy-dom, does it have advantages?
Btw `threads: false` is not a valid config for Vite 5 anymore.
`pool: 'fork'` nedds to be used.